docs/usage/design.rst

```diff
@@ -4,23 +4,121 @@ Design of SCAutolib
 This section describes design decisions that were made during
 implementation with their explanations.
 
+* **Library is designed for tests written in Beakerlib**
```

Comment:
This is not true, it is designed to be universal so it can be used in upstream (without beakerlib). Beakerlib does not impose existence of setup and cleanup phase. Setup and cleanup procedures may be used also out of beaker tests. So this paragraph is misleading. Upstream may not know what beakerlib is.

How about:
This library is designed to be compatible with RedHat testing environment that usually split test to Setup, Run and Cleanup phase. The library provides setup and cleanup procedures.

Reply:
Makes sense